xbmc/video/VideoInfoScanner.cpp

```diff
@@ -320,6 +320,22 @@ namespace VIDEO
 
       if (StringUtils::EqualsNoCase(hash, dbHash))
       { // hash matches - skipping
+
+        // see if movie set artwork empty and could be updated locally
+        if (content == CONTENT_MOVIES)
+        {
+          CVideoInfoTag movieDetails;
+          if (!items.IsEmpty() && m_database.GetMovieInfo(items[0]->GetPath(), movieDetails))
+          {
+            if (!movieDetails.m_set.title.empty())
+            {
+              CVideoInfoTag setDetails;
+              m_database.GetSetInfo(movieDetails.m_set.id, setDetails);
+              UpdateSetFromLocalArtwork(setDetails);
```

Comment:
I'm not sure that's the correct place to trigger the update during library update
1/ Why only for the case of a known dir with no change?
2/ Would be more direct (and avoid a couple db queries) to add a dedicated loop over all sets defined in the db - probably in `CVideoInfoScanner::Process()` as additional step?

Reply:
I've moved it's location in `CVideoInfoScanner::DoScan()` - it seems to fit there as the checking the directory contains `MOVIES`, `.nomedia` screening, checking the directory exists etc.. is done there and would need to be replicated in a seperate `CVideoInfoScanner::Proecss()` loop.